packages/web-forms/src/components/form-elements/ControlLabel.vue

```diff
@@ -1,12 +1,12 @@
 <script setup lang="ts">
+import MarkdownBlock from '@/components/common/MarkdownBlock.vue';
 import type { AnyControlNode as QuestionNode } from '@getodk/xforms-engine';
 const props = defineProps<{ question: QuestionNode }>();
 </script>
 
 <template>
-	<label :for="question.nodeId">
-		<span v-if="question.currentState.required" class="required">* </span>
-		<span>{{ props.question.currentState.label?.asString }}</span>
+	<label :for="question.nodeId" :class="{ required: question.currentState.required }">
```

Comment:
I've changed the `*` to be in the `:before` content because that's more semantic (the `*` is a stylistic representation of the required state, not part of the label text). It was also needed to align the `*` with rich content (eg: `<h1>`) which had some top padding.